tests/integration_tests/test_preprocessing.py

```diff
@@ -240,14 +240,47 @@ def test_read_image_from_numpy_array(tmpdir, csv_filename):
     )
 
 
+def test_read_image_failure_all_images_raise_error(monkeypatch, tmpdir, csv_filename):
+    """Tests that image feature raises an error when all images fail to be read."""
+
+    def mock_read_binary_files_no_success(self, column, map_fn, file_size):
+        """Mock read_binary_files to return None (failed image read) to test error handling."""
+        return column.map(lambda x: None)
+
+    monkeypatch.setattr(
+        ludwig.backend.base.LocalPreprocessingMixin, "read_binary_files", mock_read_binary_files_no_success
+    )
+
+    image_feature_config = image_feature(os.path.join(tmpdir, "generated_output"))
+    input_features = [image_feature_config]
+    output_features = [category_feature(decoder={"vocab_size": 5}, reduce_input="sum")]
+
+    config = {
+        "input_features": input_features,
+        "output_features": output_features,
+        TRAINER: {"epochs": 2, BATCH_SIZE: 128},
+    }
+
+    data_csv = generate_data(
+        input_features, output_features, os.path.join(tmpdir, csv_filename), num_examples=NUM_EXAMPLES, nan_percent=0.2
+    )
+
+    model = LudwigModel(config)
+
+    with pytest.raises(RuntimeError):
+        model.preprocess(data_csv)
```

Comment:
Are we guaranteed that the train set will have all NaNs? Wondering if it would be safer to bump up the nan_percent

Reply:
Ah nvm 

```
    def mock_read_binary_files_no_success(self, column, map_fn, file_size):
        """Mock read_binary_files to return None (failed image read) to test error handling."""
        return column.map(lambda x: None)
```

is clever

Reply:
Wondering if we can make do without the monkey patching and just setting nan_percent to something higher. Alternatively, maybe just using made up paths for images which can't be read to simulate this a bit further. 
Lmk what you think!

Reply:
Interesting point– setting `nan_percent` higher wouldn't necessarily work because the missing value strategy is applied before the image read step, so it may end up testing something totally different.

Using made up paths actually throws an earlier, similar error here: https://github.com/ludwig-ai/ludwig/blob/e13df72fa499c878ebef92636886e16ee0af5b83/ludwig/features/image_feature.py#L603-L606

Given the context, it seems like preprocessing for image features will actually almost always fail at that point, rather than the new code path in this PR. Curious what you think about that

Reply:
Gotcha, I didn't realize the order of steps in preprocessing images.

Let's keep it this way for now!